packages/api/tests/default_animal_breed.test.js

```diff
@@ -135,4 +155,143 @@ describe('Default Animal Breed Tests', () => {
       }
     });
   });
+
+  // POST tests
+  describe('POST custom animal breed tests', () => {
+    test('Admin users should be able to post new custom animal breed', async () => {
+      const adminRoles = [1, 2, 5];
+
+      for (const role of adminRoles) {
+        const { mainFarm, user } = await returnUserFarms(role);
+        const animal_type = await makeDefaultAnimalType();
+        const animal_breed = mocks.fakeCustomAnimalBreed({ default_type_id: animal_type.id });
+        const res = await postRequestAsPromise(animal_breed, {
+          user_id: user.user_id,
+          farm_id: mainFarm.farm_id,
+        });
+
+        // Check response
+        expect(res).toMatchObject({
+          status: 201,
+          body: {
+            farm_id: mainFarm.farm_id,
+            default_type_id: animal_breed.default_type_id,
+            breed: animal_breed.breed,
+          },
+        });
+
+        // Check database
+        const animal_breeds = await customAnimalBreedModel
+          .query()
+          .context({ showHidden: true })
+          .where('farm_id', mainFarm.farm_id)
+          .andWhere('default_type_id', animal_breed.default_type_id)
+          .andWhere('breed', animal_breed.breed);
+
+        expect(animal_breeds).toHaveLength(1);
+      }
+    });
+
+    test('Worker should not be able to post new animal custom animal breed', async () => {
+      const { mainFarm, user } = await returnUserFarms(3);
+
+      const animal_type = await makeDefaultAnimalType();
+      const animal_breed = mocks.fakeCustomAnimalBreed({ default_type_id: animal_type.id });
+      const res = await postRequestAsPromise(animal_breed, {
+        user_id: user.user_id,
+        farm_id: mainFarm.farm_id,
+      });
+
+      // Check response
+      expect(res.status).toBe(403);
+      expect(res.error.text).toBe(
+        'User does not have the following permission(s): add:animal_breeds',
+      );
+
+      const animal_breeds = await customAnimalBreedModel
+        .query()
+        .context({ showHidden: true })
+        .where('farm_id', mainFarm.farm_id)
+        .andWhere('default_type_id', animal_breed.default_type_id)
+        .andWhere('breed', animal_breed.breed);
+
+      // Check database
+      expect(animal_breeds).toHaveLength(0);
```

Comment:
I prefer not to check the database because I'd rather build the test from the client's perspective -- if I wanted to go all the way and check that the record was created, I'd do a GET request to the API and see if what I just created is returned. But that's how I think about it and not necessarily the right approach!

Reply:
I could add the get request on the specific item too. 

But maybe there is some use to the length check like if we are adding an array making sure we don't add too many (a problem with date based duplicating like copyCropPlan) and it might also check our tests are working as expected (no leftovers in db from previous tests). 

Was there a place you copied yours from @kathyavini or a specific intention here?

Reply:
I think we talked about this recently at one of those longer tech meetings, but it's just the way I've been thinking about testing controllers -- I test what they return and test if they did their purported db action, since those two are not necessarily required to be the same. I don't feel strongly about it though -- totally fine with excluding the db part.

I think I would prefer just asserting on the response over putting in a GET though -- @antsgar wouldn't that mean we wouldn't be testing the POST in isolation anymore?

Reply:
That's a good point 🙂 I'm alright with keeping the db assertions!